SoObjects/SOGo/SOGoUserManager.m

```diff
@@ -372,10 +373,22 @@ - (NSString *) getExternalLoginForUID: (NSString *) uid
         dd = [SOGoDomainDefaults defaultsForDomain: domain];
       else
         dd = [SOGoSystemDefaults sharedSystemDefaults];
-      
-      login = [dd forceExternalLoginWithEmail] ? [self getEmailForUID: uid] : uid;
+
+      if ([dd forceExternalLoginWithEmail])
+        {
+          sd = [SOGoSystemDefaults sharedSystemDefaults];
+          if ([sd enableDomainBasedUID])
+            {
+              // On multidomain environment we must use uid@domain
+              // for getEmailForUID method
+              login = [NSString stringWithFormat: @"%@@%@", uid, domain];
```

Comment:
I think we should check if UID here doesn't already have a @ - like all other checks we have.